genie-core/src/main/java/com/netflix/genie/core/services/impl/JobCoordinatorServiceImpl.java

```diff
@@ -242,15 +241,18 @@ public String coordinateJob(
                 }
             }
         } catch (GenieConflictException e) {
-            jobStateService.done(jobId);
             throw e;
```

Comment:
Lets put comments why the catches are designed like this and maybe a TODO: so when we fix the boundaries between responsibilities it's clear what we need to remove